File: src/com/antweb/silentboot/ShutdownReceiver.java
package com.antweb.silentboot;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.media.AudioManager;
import android.preference.PreferenceManager;
import android.provider.Settings;

/**
 * Handles mute on shutdown.
 * 
 * @author Anton Weber (ant@antweb.me)
 */
public class ShutdownReceiver extends BroadcastReceiver {

    /**
     * onReceive method
     *
     * @param context
     * @intent intent
     */
    @Override
    public void onReceive(Context context, Intent intent) {
        SharedPreferences settings = PreferenceManager.getDefaultSharedPreferences(context);

        if (settings.getBoolean("enabled", true)) {
            AudioManager audiomanager = (AudioManager) context
                    .getSystemService(Context.AUDIO_SERVICE);
            int lastmode = audiomanager.getRingerMode();

            SharedPreferences.Editor editor = settings.edit();
            editor.putInt("lastmode", lastmode);
            editor.commit();

            audiomanager.setRingerMode(AudioManager.RINGER_MODE_SILENT);

            if (settings.getBoolean("airplanetoggle", false)) {
                // Save current state
                int lastairplanemode = Settings.System.getInt(
                        context.getContentResolver(),
                        Settings.System.AIRPLANE_MODE_ON, 0);
                editor.putInt("lastairplanemode", lastairplanemode);

                // Deactivate airplane mode if necessary
                if (lastairplanemode != 0) {
                    Settings.System.putInt(context.getContentResolver(),
                            Settings.System.AIRPLANE_MODE_ON, 0);
                }
            }
        }
    }
}

File: app/src/main/kotlin/com/antweb/silentboot/ShutdownReceiver.kt
package com.antweb.silentboot

import android.app.NotificationManager
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.media.AudioManager
import androidx.preference.PreferenceManager

class ShutdownReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val sharedPrefs = PreferenceManager.getDefaultSharedPreferences(context)
        val isEnabled = sharedPrefs.getBoolean(PreferenceKey.ENABLED.key, true)

        if (!isEnabled) {
            return
        }

        val audioManager = context
            .getSystemService(Context.AUDIO_SERVICE) as AudioManager
        val notificationManager = context
            .getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager

        val lastRingerMode = audioManager.ringerMode
        val lastDnd = notificationManager.currentInterruptionFilter

        val editor = sharedPrefs.edit()
        editor.putInt(PreferenceKey.LAST_RINGER_MODE.key, lastRingerMode)
        editor.putInt(PreferenceKey.LAST_DND_MODE.key, lastDnd)
        editor.apply()

        audioManager.ringerMode = AudioManager.RINGER_MODE_SILENT
        notificationManager.setInterruptionFilter(NotificationManager.INTERRUPTION_FILTER_NONE)

        // Stick to commit to be on the safe side?
        editor.commit()
    }
}
